mediapipe/tasks/ios/vision/gesture_recognizer/sources/MPPGestureRecognizer.mm

```diff
@@ -0,0 +1,273 @@
+                                                                        .cppString]
+                                   handednessPacket:outputPacketMap[kHandednessOutStreamName
+                                                                        .cppString]
+                                handLandmarksPacket:outputPacketMap[kLandmarksOutStreamName
+                                                                        .cppString]
+                               worldLandmarksPacket:outputPacketMap[kWorldLandmarksOutStreamName
+                                                                        .cppString]];
+}
+
+- (instancetype)initWithOptions:(MPPGestureRecognizerOptions *)options error:(NSError **)error {
+  self = [super init];
+  if (self) {
+    MPPTaskInfo *taskInfo = [[MPPTaskInfo alloc]
+        initWithTaskGraphName:kTaskGraphName
+                 inputStreams:@[
+                   [NSString stringWithFormat:@"%@:%@", kImageTag, kImageInStreamName],
+                   [NSString stringWithFormat:@"%@:%@", kNormRectTag, kNormRectInStreamName]
+                 ]
+                outputStreams:@[
+                  [NSString stringWithFormat:@"%@:%@", kLandmarksTag, kLandmarksOutStreamName],
+                  [NSString
+                      stringWithFormat:@"%@:%@", kWorldLandmarksTag, kWorldLandmarksOutStreamName],
+                  [NSString stringWithFormat:@"%@:%@", kHandednessTag, kHandednessOutStreamName],
+                  [NSString
+                      stringWithFormat:@"%@:%@", kHandGesturesTag, kHandGesturesOutStreamName],
+                  [NSString stringWithFormat:@"%@:%@", kImageTag, kImageOutStreamName]
+                ]
+                  taskOptions:options
+           enableFlowLimiting:options.runningMode == MPPRunningModeLiveStream
+                        error:error];
+
+    if (!taskInfo) {
+      return nil;
+    }
+
+    PacketsCallback packetsCallback = nullptr;
+
+    if (options.gestureRecognizerLiveStreamDelegate) {
+      _gestureRecognizerLiveStreamDelegate = options.gestureRecognizerLiveStreamDelegate;
+      // Capturing `self` as weak in order to avoid `self` being kept in memory
+      // and cause a retain cycle, after self is set to `nil`.
+      MPPGestureRecognizer *__weak weakSelf = self;
+
+      // Create a private serial dispatch queue in which the deleagte method will be called
+      // asynchronously. This is to ensure that if the client performs a long running operation in
+      // the delegate method, the queue on which the C++ callbacks is invoked is not blocked and is
+      // freed up to continue with its operations.
+      const char *queueName = [MPPVisionTaskRunner uniqueDispatchQueueNameWithSuffix:kTaskName];
+      dispatch_queue_t callbackQueue = dispatch_queue_create(queueName, NULL);
+      packetsCallback = [=](absl::StatusOr<PacketMap> status_or_packets) {
+        if (!weakSelf) {
```

Comment:
I just got some feedback regarding this style. Can you extract a helper method similar to FaceDetector?

The reviewer comment:

"Default capture by value should only be used for very short lambdas, and this one's over 50 lines. Prefer to explicitly capture weakSelf and callbackQueue.

Alternately, since the entire lambda body depends on weakSelf being non-nil, what you might want to do is move the body of this fxn to a private member function on MPPFaceDetector, and have the lambda just be a thunk to the ObjC++ method, e.g."

Reply:
I have updated this for the gesture recogniser. But please check this comment regarding the face detector.
https://github.com/google/mediapipe/pull/4464/files#r1205695278